types/styled-components/test/index.tsx

```diff
@@ -946,3 +946,41 @@ function validateArgumentsAndReturns() {
     // $ExpectError
     createGlobalStyle([]);
 }
+
+function validateDefaultProps() {
+    interface Props {
+        requiredProp: boolean;
+        optionalProp: string; // Shouldn't need to be optional here
+    }
+
+    class MyComponent extends React.PureComponent<Props> {
+        static defaultProps = {
+            optionalProp: 'fallback'
+        };
+
+        render() {
+            const { requiredProp, optionalProp } = this.props;
+            return (
+                <span>
+                    {requiredProp.toString()}
+                    {optionalProp.toString()}
+                </span>
+            );
+        }
+    }
+
+    const StyledComponent = styled(MyComponent)`
+        color: red
+    `;
+
+    // this test is failing in TS 2.9 but not in 3.0
+    // <MyComponent requiredProp />;
+
+    <StyledComponent requiredProp optionalProp="x" />;
+
+    // this test is failing in TS 3.0 but not in 3.1
+    // <StyledComponent requiredProp />;
```

Comment:
these tests are failing on 2.9 and 3.0.
Is there a way to ignore them only on specific versions of TS?

Reply:
You can create new definitions for ts 3.2+ version

Reply:
Yeah, this can only pass in TS >= 3.2. TS < 3.2 had a bug where it would _only_ apply `defaultProps` on class components, and styled components aren't class components.

Reply:
I think @sheetalkamat meant to use [a new `typesVersions` TS 3.1 feature](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html#version-selection-with-typesversions), if I'm not mistaken.

I could do that but the issue has nothing to do with definition itself, and only appears in the tests. So if were a way to have tests for specific versions of TS that would help here.

I am not sure if we already have test/build infrastructure that allows to leverage the new feature.

My personal opinion, that since there's no real breaking changes in the definition, the minimal TS version should not be advanced. The tests, however, might have different expectations depending on TS version. And that's where we should look at.